### DISCUSSION TITLE
Vertical shell settings

### DISCUSSION THREAD

**Discussion by Carib3D** [supermerill/SuperSlicer#2588] [Q&A] [answered]
I need to design a large component with a lot of vertical walls so I'm trying to make the width as efficient as possible but I'm not able to get what I need. In this case, I'm using a 0.6 nozzle, 0.25 layer height and believe that two perimeters should be sufficient. Looking at the vertical shell information below suggests that I should be setting the wall thickness to 2.46 mm.
![VS1](https://user-images.githubusercontent.com/36038546/160435116-4fe8b5ee-0c48-4bea-8a67-58585e03180a.png)
Doing a slicer check on a test piece, it generates 4 perimeters, as shown below, rather than the 2 expected. I can't see any way of correcting this as I think 4 thin perimeters will be comparatively weak and take longer to print. Any suggestions?
![VS2](https://user-images.githubusercontent.com/36038546/160436048-1a3d0621-efd5-4c1c-9ab2-62783fee1f8a.png)

**Comment by supermerill** [comment, marked as answer]
You are looking at two perimeter loops in the picture.
The number indicated is the one for the field above.
"if you set X perimeter (in the field above), then it can print a wall of min thinkness of Ymm". If you have ideas for a better sentence, tell me.

If you want to tune your extrusion width instead of the object wall, look at the 2.4 beta (the nightly is the less buggy), as it has a field to automatically compute & change it.

**Reply by Carib3D** [reply]
Got it! The field gives number of perimeters per side. I did think that 2.46 was very wide for 2 walls, but 1.21 was too narrow. I'll play with extrusion width options in the current version and maybe look at the 2.4 beta a bit later but I've been holding out for the stable version. Thanks for your quick response!
